### PR TITLE
Refactored NotificationService

### PR DESCRIPTION
**Pull request by DynxstyGIT**

### CODE REVIEW

File: src/main/java/net/javadiscord/javabot/systems/notification/GuildNotificationService.java [hunk 1]
```diff
@@ -1,62 +1,50 @@
...
+import java.util.function.Function;
 
 /**
- * Sends notifications within a single {@link Guild}.
+ * Handles all sorts of guild notifications.
  */
 @Slf4j
-public final class GuildNotificationService extends NotificationService {
+@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
+public final class GuildNotificationService extends NotificationService.MessageChannelNotification {
```
Comment (danthe1st) [inline src/main/java/net/javadiscord/javabot/systems/notification/GuildNotificationService.java:19]: Maybe make this class package private?

File: src/main/java/net/javadiscord/javabot/systems/notification/NotificationService.java [hunk 2]
```diff
@@ -1,35 +1,59 @@
...
+
+	public static @NotNull QOTWNotificationService withQOTW(Guild guild, User user) {
+		return new QOTWNotificationService(user, guild);
+	}
+
+	/**
+	 * Abstract class which streamlines the logic of sending messages to a {@link MessageChannel}.
+	 */
+	@Slf4j
+	protected abstract static class MessageChannelNotification {
```
Comment (danthe1st) [inline src/main/java/net/javadiscord/javabot/systems/notification/NotificationService.java:43]: Why is this class `protected` if no subclasses of `NotificationService ` are allowed?
Wouldn't package private be sufficient?

File: src/main/java/net/javadiscord/javabot/systems/notification/NotificationService.java [hunk 3]
```diff
@@ -1,35 +1,59 @@
...
+ * Handles all types of guild & user notifications.
  */
-@Slf4j
-public abstract class NotificationService {
-	void sendDirectMessageNotification(User user, MessageEmbed message) {
-		user.openPrivateChannel().queue(
-				channel -> sendMessageChannelNotification(channel, message),
-				error -> log.warn("Could not send private Notification to User " + user.getAsTag())
-		);
+public class NotificationService {
```
Comment (danthe1st) [inline src/main/java/net/javadiscord/javabot/systems/notification/NotificationService.java:17]: Make this class `final` to document the fact that is is not possible to create subclasses of `NotificationService` (due to the private constructor)

**Review by danthe1st — CHANGES_REQUESTED** [review]

File: src/main/java/net/javadiscord/javabot/systems/notification/GuildNotificationService.java (hunk 1 above)
Comment (DynxstyGIT) [reply src/main/java/net/javadiscord/javabot/systems/notification/GuildNotificationService.java:19]: Why? This wouldn't let me access the classes' methods from the outside

Comment (danthe1st) [reply src/main/java/net/javadiscord/javabot/systems/notification/GuildNotificationService.java:19]: I forgot to delete that comment after looking at the other classes.

**Review by danthe1st — APPROVED** [review]

**Pull request merged**